src/components/FormElements/FetchContractMethodPickerWithQuery.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from "react";
+import { Button } from "@stellar/design-system";
+
+import { Box } from "@/components/layout/Box";
+import { ContractMethodSelectPicker } from "@/components/FormElements/ContractMethodSelectPicker";
+import { TextPicker } from "@/components/FormElements/TextPicker";
+import { MessageField } from "@/components/MessageField";
+
+import {
+  fetchContractFunctionMethods,
+  ContractFunctionMethods,
+} from "@/helpers/sorobanUtils";
+
+import { useStore } from "@/store/useStore";
+import { validate } from "@/validate";
+
+interface FetchContractMethodPickerWithQueryProps {
+  id: string;
+  value: string;
+  error?: string | undefined;
+  label?: string;
+  disabled?: boolean;
+  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+/**
+ * FetchContractMethodPickerWithQuery
+ *
+ * This component is used to fetch the contract methods for a given contract id.
+ * It is used in the SorobanOperation component.
+ *
+ */
+export const FetchContractMethodPickerWithQuery = ({
+  id,
+  value,
+  error,
+  label,
+  disabled,
+  onChange,
+}: FetchContractMethodPickerWithQueryProps) => {
+  const { network } = useStore();
+
+  const [contractIdError, setContractIdError] = useState<string>("");
+  const [contractMethods, setContractMethods] = useState<string[]>([]);
+  const [fetchError, setFetchError] = useState<string>("");
+
+  const onContractIdChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    // reset the error and methods
+    setFetchError("");
+    setContractMethods([]);
+
+    // call the onChange callback
+    onChange?.(e);
+
+    // validate the contract id
+    if (e.target.value) {
+      const validatedContractIdError = validate.getContractIdError(
+        e.target.value,
+      );
+
+      if (validatedContractIdError) {
+        setContractIdError(validatedContractIdError);
+      } else {
+        setContractIdError("");
+      }
+    }
+  };
+
+  const handleFetchContractMethods = async () => {
+    const contractMethods: ContractFunctionMethods =
```

Comment:
We should clear the error here as well. I got an error with one RPC URL, then changed to another that worked, but the error wasn't cleared.